portality/api/v1/discovery.py

```diff
@@ -301,72 +294,17 @@ def __init__(self, qs, fro, psize, sortby=None, sortdir=None):
     def query(self):
         q = {
             "query" : {
-                "filtered" : {
-                    "filter" : {
-                        "bool" : {
-                            "must" : [
-                                {"term" : {"admin.in_doaj": True}}
-                            ]
-                        }
-                    },
-                    "query" : {
-                        "query_string" : {
-                            "query" : self.qs,
-                            "default_operator": "AND"
-                        }
-                    }
+                "query_string" : {
+                    "query" : self.qs,
+                    "default_operator": "AND"
                 }
             },
-            "_source": {
-                "include": ["last_updated", "created_date", "id", "bibjson"],
-                "exclude": [],
-            },
             "from" : self.fro,
             "size" : self.psize
         }
-
-        if self.sortby is not None:
-            q["sort"] = [{self.sortby : {"order" : self.sortdir, "mode" : "min"}}]
-
-        return q
-
-class ApplicationQuery(object):
-    def __init__(self, owner, qs, fro, psize, sortby=None, sortdir=None):
-        self.owner = owner
-        self.qs = qs
-        self.fro = fro
-        self.psize = psize
-        self.sortby = sortby
-        self.sortdir = sortdir if sortdir is not None else "asc"
-
-    def query(self):
-        q = {
-            "query" : {
-                "filtered" : {
-                    "filter" : {
-                        "bool" : {
-                            "must" : [
-                                {"term" : {"admin.owner.exact": self.owner}}
-                            ]
-                        }
-                    },
-                    "query" : {
-                        "query_string" : {
-                            "query" : self.qs,
-                            "default_operator": "AND"
-                        }
-                    }
-                }
-            },
-            "_source": {
-                "include": ["admin.application_status", "suggestion", "last_updated", "created_date", "id", "bibjson"],
-                "exclude": [],
-            },
-            "from" : self.fro,
-            "size" : self.psize
-        }
-
         if self.sortby is not None:
             q["sort"] = [{self.sortby : {"order" : self.sortdir, "mode" : "min"}}]
 
-        return q
\ No newline at end of file
+        query_service = DOAJ.queryService()
+        query, dao_klass = query_service.prepare_query('api_query', self.index_type, q, self.account)
```

Comment:
After pondering it for a bit, I don't think this bit of code should be in this place.  The idea of the Query objects is they are a bit like Prepared Statements in SQL and by adding a second part to the return, and an integration with the queryService, it feels like this class is over-stretching its intention.  Instead I would have this class return the pure search query that it *wants* to run, and then in the caller (`_make_search`) add this call to the query service to constrain the query and get the class.

Reply:
Although, see my comments about not using `prepare_query` in the first place too :)